arangod/Replication2/StateMachines/Document/MaintenanceActionExecutor.cpp

```diff
@@ -122,6 +124,25 @@ auto MaintenanceActionExecutor::executeCreateIndex(
   return res;
 }
 
+auto MaintenanceActionExecutor::executeDropIndex(ShardID shard,
+                                                 velocypack::SharedSlice index)
+    -> Result {
+  auto col = _vocbase.lookupCollection(shard);
```

Comment:
If my comment on the DataBaseGuard is correct then we need the guard here as well.